tiledb/sm/c_api/tiledb.cc

```diff
@@ -2609,14 +2671,22 @@ int tiledb_kv_item_get_value(
     return TILEDB_ERR;
 
   if (attribute == nullptr) {
-    auto st = tiledb::sm::Status::Error(
+    auto st = tiledb::sm::Status::KVItemError(
```

Comment:
I think we should keep the generic `Error` in `tiledb.cc`. The format we use is typically `<Class>Error` when the error happens inside the actual class `<Class>` (there are some exceptions, like `IOError` that intend to fix).